src/app/gallery/checkout/page.tsx

```diff
@@ -0,0 +1,67 @@
+'use client'
+import { useEffect, useState } from 'react';
+import { Button } from '~/components/ui/button';
+import { stripePromise } from './load-stripe';
+import { Alert, AlertDescription, AlertTitle } from "~/components/ui/alert"
+
+
+
+export default function PreviewPage() {
+
+    // eslint-disable-next-line
+    const stripe = stripePromise;
+
+    const [isSuccess, setIsSuccess] = useState(false);
+    const [isCanceled, setIsCanceled] = useState(false);
+
+
+    useEffect(() => {
+        // NOTE: Check to see if this is a redirect back from Checkout.
+        // NOTE: `window` object only exists on the client-side, if it is used outside the 
+        // useEffect() would cause 'undefined', because of running on server-side rendering.
+        // So, it needs to call useState() first.
+        const query = new URLSearchParams(window.location.search);
```

Comment:
Very much check why this would be the case - this code SHOULD be on the client, so why does it give you an error that should only be on the server?

Also remember Node has the `?` operator to execute if an object exists, which can be handy in dealing with `undefined`

Reply:
I checked, this file runs on the client side.